avm/res/compute/disk-encryption-set/modules/nested_keyVaultPermissions.bicep

```diff
@@ -0,0 +1,69 @@
+@description('Required. A boolean to specify whether or not the used Key Vault has RBAC authentication enabled or not.')
+param rbacAuthorizationEnabled bool = true
+
+@description('Required. The resourceID of the User Assigned Identity to assign permissions to.')
+param userAssignedIdentityResourceId string
+
+@description('Optional. Resource location.')
+param location string = resourceGroup().location
+
+@description('Required. Resource ID of the KeyVault containing the key or secret.')
+param keyVaultResourceId string
+
+@description('Required. Key URL (with version) pointing to a key or secret in KeyVault.')
+param keyName string
+
+resource keyVault 'Microsoft.KeyVault/vaults@2021-10-01' existing = {
+  name: last(split(keyVaultResourceId, '/'))!
+
+  resource key 'keys@2021-10-01' existing = {
+    name: keyName
+  }
+}
+
+module userAssignedIdentity 'nested_managedIdentityReference.bicep' = {
+  name: '${uniqueString(deployment().name, location)}-MSI-Reference'
+  params: {
+    userAssignedIdentityName: last(split(userAssignedIdentityResourceId, '/'))!
+    location: location
+  }
+  scope: resourceGroup(split(userAssignedIdentityResourceId, '/')[2], split(userAssignedIdentityResourceId, '/')[4])
+}
+
+// =============== //
+// Role Assignment //
+// =============== //
+
+resource keyVaultKeyRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (rbacAuthorizationEnabled == true) {
```

Comment:
```suggestion
resource keyVaultKeyRBAC 'Microsoft.Authorization/roleAssignments@2022-04-01' = if (keyVault.properties.enableRbacAuthorization == true) {
```

Reply:
Same as below: 
Unfortunately, this doesn't work and we probably need to keep the `rbacAuthorizationEnabled` parameter. 
The reason:
`This expression is being used in the if-condition expression, which requires a value that can be calculated at the start of the deployment.`